### ISSUE TITLE
RebuildIndex shows indexComplete message too early

### ISSUE THREAD

**Issue by fritzmg** [contao/core#8262]
## Problem

When you are rebuilding the search index in the maintenance section, the `indexComplete` message is sometimes shown too early, before all requests for all pages are finished. It's more likely to happen if you either have a lot of pages or at least some pages that take a long time to generate.

## Reproduction

1. create a new file named `sleep.php` in your `/templates` directory and insert the following content: `<?php session_write_close(); sleep(25);` - change the number of seconds to sleep to something sufficiently high, if necessary.
2. create a new HTML content element on your index page for example and insert `{{file::sleep.php}}`. If you want, add this script to a few other pages as well.
3. go to _System - Maintenance_ and click on "Rebuild index"
4. The message "The search index has been rebuilt. You can now proceed." will be shown, even though not all pages have been processed yet.

![si](https://cloud.githubusercontent.com/assets/4970961/13598793/19212d9c-e520-11e5-9255-cab799b3ee9b.png)

## Cause

The logic of the MooTools script shows the message once the request of the __last element__ in the list of URLs is completed.
```JavaScript
          if (i == last) {
            $('index_loading').setStyle('display', 'none');
            $('index_complete').setStyle('display', 'block');
          }
```
However, the last request in the list of URLs is not necessarily the last request to be finished overall. The requests could be executed out of order by the Browser and, as shown with the reproduction above, some pages may take longer to generate, e.g. longer than the request of the last element in the list.

## Fix

Exchange the [current script](https://github.com/contao/core/blob/master/system/modules/core/templates/backend/be_rebuild_index.html5#L19) in `be_rebuild_index.html5` with this:
```JavaScript
<script>
  window.addEvent('domready', function() {
    var urls = $$('.page_url'),
        complete = 0;
    urls.each(function(el, i) {
      el.getAttribute('data-url') && new Request({
        'url': el.getAttribute('data-url'),
        onComplete: function() {
          el.addClass('tl_green');
          complete++;
          if (complete == urls.length) {
            $('index_loading').setStyle('display', 'none');
            $('index_complete').setStyle('display', 'block');
          }
        }
      }).get();
    });
  });
</script>
```
Or, while you are at it, use the script from #8255 which introduces more feedback.

**Comment by leofeyer** [comment]
Good catch!

**Comment by leofeyer** [comment]
Fixed in 266460865770e785786901c2ff33e59c7249bc27.
